keras_retinanet/utils/coco_eval.py

```diff
@@ -34,29 +34,23 @@ def evaluate_coco(generator, model, threshold=0.05):
         image, scale = generator.resize_image(image)
 
         # run network
-        _, _, detections = model.predict_on_batch(np.expand_dims(image, axis=0))
-
-        # clip to image shape
-        detections[:, :, 0] = np.maximum(0, detections[:, :, 0])
-        detections[:, :, 1] = np.maximum(0, detections[:, :, 1])
-        detections[:, :, 2] = np.minimum(image.shape[1], detections[:, :, 2])
-        detections[:, :, 3] = np.minimum(image.shape[0], detections[:, :, 3])
+        _, _, boxes, classification = model.predict_on_batch(np.expand_dims(image, axis=0))
```

Comment:
Yeah true, the classification comes before the boxes, but the last output is the classification scores after NMS. I'm thinking we should change the name to avoid this confusion ... how about `nms_classification`?

Reply:
Sounds good.

Also, by moving the box clipping into the network it now happens before nms, while before it happened after nms. I'm not sure which is better, or if it really matters, but it may be worth thinking about the order.

Reply:
Good point, but I'd say it is better.

Small example: assume that after clipping two boxes are identical and touching one of the sides of the image, but before clipping one of the two boxes extends far beyond that side of the image. The two boxes should have a high overlap in my opinion since we ignore area's outside of the image. If we do NMS before clipping both boxes would survive, after clipping one box would be suppressed.